tests/authentication/test_auth_change_password.py

```diff
@@ -28,41 +38,411 @@ class TestResetPassword:
+            ),
+            pytest.param(
+                "testtest",
+                None,
+                generate_password(8),
+                400,
+                "Incorrect token format, token must be ###-###-###",
+            ),
+            pytest.param(
+                "123-123-123 ", None, generate_password(8), 400, "Incorrect token"
+            ),
+            pytest.param(
+                "12345678910 ",
+                None,
+                generate_password(8),
+                400,
+                "Incorrect token format, token must be ###-###-###",
+            ),
+            pytest.param(
+                "$*@!_+_*&%",
+                None,
+                generate_password(8),
+                400,
+                "Incorrect token format, token must be ###-###-###",
+            ),
+        ],
+    )
+    def test_reset_password_incorrect_code(
+        self,
+        create_authorized_user,
+        code_for_reset_password,
+        email,
+        new_password,
+        expected_status_code,
+        expected_message_part,
+    ):
+        with step("Registration user"):
+            user = create_authorized_user["user"]
+
+        with step("Sent request to reset password"):
+            email = user["email"] if email is None else email
+            response_reset_password = AuthenticateAPI().change_password_through_reset(
+                email=email,
+                code_for_reset_password=code_for_reset_password,
+                new_password=new_password,
+                expected_status_code=400,
+            )
+
+        with step("Verify error message"):
+            assert_message_in_response(
+                response=response_reset_password, expected_message=expected_message_part
+            )
+
+        with step(
+            "Verify password does not reset successfully by Authorization request with new password"
+        ):
+            AuthenticateAPI().authentication(
+                email=email, password=new_password, expected_status_code=401
+            )
+
+    @pytest.mark.xfail(
+        reason="Incorrect error message and status code."
+        "Link: https://trello.com/c/MiA5Evtu/17-reset-password-with-invalid-email-unclear-error-message"
+    )
+    @pytest.mark.medium
+    @severity(severity_level="MEDIUM")
+    @title("Test change password through reset password, incorrect email")
+    @description(
+        "Give user registered."
+        "WHEN user sent request to change password with incorrect code."
+        "THEN status HTTP CODE = 400 and appropriate error message"
+    )
+    @pytest.mark.parametrize(
+        "code_for_reset_password,email, new_password, expected_status_code, expected_message_part",
+        [
+            pytest.param(
+                " ",
+                "example.domain.com",
+                generate_password(8),
+                400,
+                "Email must be valid",
```

Comment:
This message is repeated, i would extract it to a constant, e.g. `INCORRECT_EMAIL_MSG=...`

Reply:
fixed